<launch>
    <arg name="custom_map_server" default="1"/>
    <arg name="rviz_config" default="$(find rrt_planner)/cfg/rrt_planner.rviz"/>
    <group if="$(eval arg('custom_map_server') ==0)">
        <arg name="map_file" default="$(find rrt_planner)/cfg/map.yaml"/>
        <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>
    </group>
    <group if="$(eval arg('custom_map_server') ==1)">
        <node type="map_server.py" pkg="rrt_planner" name="map_server" output="screen"/>
    </group>

    <rosparam command="load" file="$(find rrt_planner)/cfg/config.yaml"/>
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rviz_config)"/>
    <node type="rrt_planner_node" name="rrt_planner" pkg="rrt_planner" output="screen"/>
</launch>